fixtures/coverall/tests/bindings/test_handlerace.kts

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import java.util.concurrent.*
+import uniffi.coverall.*
+
+// This tests the thread-safety of operations on object handles.
+// The idea is to have a method call running on one thread while
+// we try to destroy the object on another thread. This should
+// be handled correctly at the Kotlin level, rather than triggering
+// any fail-safe checks (or worse, memory unsafety!) in the underlying
+// Rust code.
+//
+// This test was able to reproduce the concurrency issue in
+// https://github.com/mozilla/uniffi-rs/issues/457, and is kept
+// to prevent regressions.
```

Comment:
Ref https://github.com/mozilla/uniffi-rs/issues/457#issuecomment-846747337 for the failure that was previously being demonstrated by this test.